
in linux
sudo apt update
sudo apt upgrade


rm -rf venv
python3 -m venv venv
source venv/bin/activate
pip install django
pip install psycopg2-binary
pip install dj_database_url
pip install whitenoise
pip install python-dotenv
pip install python-dateutil
pip install gunicorn

pip freeze  requirements.txt

or pip install -r requirements.txt


django-admin startproject PROJECT .
python manage.py startapp APP

in settings:

import os
import dj_database_url
from dotenv import load_dotenv
import django.conf.locale

ALLOWED_HOSTS = ['127.0.0.1','0.0.0.0','localhost','forecast-21v1.onrender.com','86.247.69.203']

SECURE_SSL_REDIRECT = os.getenv('SECURE_SSL_REDIRECT','False') == True

LOGIN_REDIRECT_URL = '/'
LOGIN_URL = '/accounts/login/'
LOGOUT_REDIRECT_URL = '/'
EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
EMAIL_HOST = 'smtp.gmail.com'
EMAIL_FROM = 'thierry.fanahafa@gmail.com'
EMAIL_HOST_USER = 'thierry.fanahafa@gmail.com'
EMAIL_HOST_PASSWORD = 'staidshlprizhgfk'
EMAIL_PORT = 587
EMAIL_USE_TLS = True

middleware admin css corrupt if there is no whitenoise

 'whitenoise.middleware.WhiteNoiseMiddleware',   
    'django.middleware.locale.LocaleMiddleware',
    
static 



STATIC_URL = '/static/'
STATIC_ROOT = os.path.join(BASE_DIR,'staticfiles')

STATICFILES_DIRS = [
    ("APP", BASE_DIR / "APP/static/",),
]




python manage.py makemigrations
python manage.py migrate
python manage.py createsuperuser
python manage.py collectstatic
python manage.py runserver 0.0.0.0:8004

in settings 

in windows 
.\.venv\Scripts\activate 

gunicorn ATHB.wsgi:application --bind 0.0.0.0:8004

sudo mkdir -p /var/log/ATHB
sudo chown -R tita:www-data /var/log/ATHB

sudo nano /etc/systemd/system/ATHB.service


[Unit]
Description=Gunicorn daemon for Django app
After=network.target

[Service]
User=tita
Group=www-data
WorkingDirectory=/home/tita/Documents/DEV/ATHOMEBEAUTY
Environment="PATH=/home/tita/Documents/DEV/ATHOMEBEAUTY/venv/bin"
ExecStart="/home/tita/Documents/DEV/ATHOMEBEAUTY/venv/bin/gunicorn" ATHB.wsgi:application --bind 0.0.0.0:8004 \
  --timeout 120 \
  --access-logfile /var/log/ATHB/access.log \
  --error-logfile /var/log/ATHB/error.log 

[Install]
WantedBy=multi-user.target



sudo systemctl daemon-reexec
sudo systemctl daemon-reload
sudo systemctl enable ATHB.service

sudo systemctl start ATHB.service
sudo systemctl status ATHB.service


sudo nano /etc/nginx/sites-available/ATHB

server {
    listen 8082;
    server_name localhost;

    location / {
        proxy_pass http://127.0.0.1:8004;
        include proxy_params;
        proxy_redirect off;
    }
}

sudo ln -s /etc/nginx/sites-available/ATHB /etc/nginx/sites-enabled/
sudo nginx -t
sudo systemctl reload nginx


sudo nginx -t
sudo systemctl restart nginx

sudo systemctl enable nginx
sudo systemctl enable ATHB

sudo ufw allow 8082/tcp
sudo ufw enable
sudo ufw status

to check if nginx running 

sudo lsof -i :8082

if something is running

sudo systemctl stop service_concerné
sudo systemctl disable service_concerné



sudo systemctl restart ATHB

for internalization
 in settings
 import django.conf.locale
 
Available languages
LANGUAGES = [
    ('en', 'English'),
    ('fr', 'Français'),
    ('mg', 'Malagasy'), 
]

LOCALE_PATHS = [BASE_DIR / 'locale']


 in models or view
  from django.utils.translation import gettext_lazy as _

in template
{% load i18n %}
{% trans "Home" %}


sudo apt-get install gettext
django-admin makemessages -l fr
django-admin makemessages -l mg
django-admin compilemessages